recipe/run_test.sh

```diff
@@ -0,0 +1,8 @@
+# Skip compile tests if we are cross-compiling
+if echo ${CONDA_TOOLCHAIN_HOST} | grep -q arm64-apple; then
```

Comment:
```suggestion
if [[ "${CONDA_BUILD_CROSS_COMPILATION:-0}" == "1" && "${CMAKE_CROSSCOMPILING_EMULATOR:-}" == "" ]]; then
```

Reply:
I had something like that earlier but it didn't work - looks like `activate_clang_osx-arm64.sh` unsets `CONDA_BUILD_CROSS_COMPILATION` when run in the test environment.

Reply:
That's weird, I'm using exactly this in the `go-activation-feedstock`.

Reply:
Indeed, you can see the failure at https://github.com/conda-forge/boost-cpp-feedstock/runs/5082034667 (and the corresponding [run_test.sh](https://github.com/conda-forge/boost-cpp-feedstock/blob/e0a309637954cb4121298ed058d869b8aaf198f1/recipe/run_test.sh)). I can't see the Azure Pipelines logs right now (says "An unexpected error has occurred within this region of the page") but maybe it'll work for you. I'm certainly open to suggestions!